src/liballoc/collections/linked_list.rs

```diff
@@ -986,6 +1080,321 @@ impl<T> IterMut<'_, T> {
+            let unlinked_node = Box::from_raw(unlinked_node.as_ptr());
+            Some((*unlinked_node).element)
+        }
+    }
+
+    /// Insert `list` between the current element and the next
+    #[unstable(feature = "linked_list_cursors", issue = "58533")]
+    pub fn splice_after(&mut self, list: LinkedList<T>) {
+        unsafe {
+            let (splice_head, splice_tail, splice_len) = match list.detach_all_nodes() {
+                Some(parts) => parts,
+                _ => return,
+            };
+            let (node, node_next) = match self.current {
+                None => (None, self.list.head),
+                Some(node) => {
+                    let node_next = node.as_ref().next;
+                    (Some(node), node_next)
+                }
+            };
+            self.list.splice_nodes(node, node_next, splice_head, splice_tail, splice_len);
+            if self.current.is_none() {
+                // The ghost element's index has increased by `splice_len`.
+                self.index += splice_len;
+            }
+        }
+    }
+
+    /// Insert `list` between the previous element and current
+    #[unstable(feature = "linked_list_cursors", issue = "58533")]
+    pub fn splice_before(&mut self, list: LinkedList<T>) {
+        unsafe {
+            let (splice_head, splice_tail, splice_len) = match list.detach_all_nodes() {
+                Some(parts) => parts,
+                _ => return,
+            };
+            let (node_prev, node) = match self.current {
+                None => (self.list.tail, None),
+                Some(node) => {
+                    let node_prev = node.as_ref().prev;
+                    (node_prev, Some(node))
+                }
+            };
+            self.list.splice_nodes(node_prev, node, splice_head, splice_tail, splice_len);
+            self.index += splice_len;
+        }
+    }
+
+    /// Split the list in two after the current element
+    /// The returned list consists of all elements following the current one.
+    // note: consuming the cursor is not necessary here, but it makes sense
+    // given the interface
+    #[unstable(feature = "linked_list_cursors", issue = "58533")]
+    pub fn split_after(self) -> LinkedList<T> {
+        let split_off_idx = if self.index == self.list.len { 0 } else { self.index + 1 };
+        unsafe {
+            let split_off_node = self.current;
+            self.list.split_off_after_node(split_off_node, split_off_idx)
+        }
+    }
+    /// Split the list in two before the current element
+    #[unstable(feature = "linked_list_cursors", issue = "58533")]
+    pub fn split_before(self) -> LinkedList<T> {
+        let split_off_idx = self.index;
+        unsafe {
+            let split_off_node = match self.current {
+                Some(node) => node.as_ref().prev,
+                None => self.list.tail,
+            };
+            self.list.split_off_after_node(split_off_node, split_off_idx)
```

Comment:
It would be slightly more efficient to have a `split_off_before_node` method.

Reply:
done. also changed behavior, now returning the "first part" as return value